test/unit/system_uri_test.py

```diff
@@ -61,6 +61,30 @@ def test_translate_obs_uri_not_found(mock_request_get, self):
         uri.runtime_config = self.runtime_config
         assert uri.translate()
 
+    @raises(KiwiUriOpenError)
+    @patch('kiwi.system.uri.Defaults.is_buildservice_worker')
+    def test_translate_obs_uri_in_private_buildservice(
+            mock_buildservice, self
+    ):
+        mock_buildservice.return_value = True
+        self.runtime_config.is_obs_public = mock.Mock(
+            return_value=False
+        )
+        self.runtime_config.get_obs_download_server_url = mock.Mock(
+            return_value='obs_server'
+        )
```

Comment:
The test works, but you'll see that the coverage is not complete, thus it does not perform as expected. An exception is thrown, but not the one that is supposed to happen.